test/axes/baseAxisTests.ts

```diff
@@ -206,4 +206,615 @@ describe("BaseAxis", () => {
+      });
+
+      it("can set if annotations are enabled", () => {
+        let axis = new Plottable.Axis(new Plottable.Scale<{}, number>(), "bottom");
+        let annotationsEnabled = axis.annotationsEnabled();
+        axis.annotationsEnabled(!annotationsEnabled);
+        assert.strictEqual(axis.annotationsEnabled(), !annotationsEnabled, "can set if annotations are enabled");
+      });
+
+      it("returns the axis when setting if annotations are enabled", () => {
+        let axis = new Plottable.Axis(new Plottable.Scale<{}, number>(), "bottom");
+        assert.strictEqual(axis.annotationsEnabled(true), axis, "enabling annotations return calling axis");
+      });
+    });
+
+    describe("annotating ticks", () => {
+      it("has no annotated ticks by default", () => {
+        let axis = new Plottable.Axis(new Plottable.Scale<{}, number>(), "bottom");
+        assert.deepEqual(axis.annotatedTicks(), [], "no annotated ticks by default");
+      });
+
+      it("can set the annotated ticks", () => {
+        let axis = new Plottable.Axis(new Plottable.Scale<{}, number>(), "bottom");
+        let annotatedTicks = [20, 30, 40];
+        axis.annotatedTicks(annotatedTicks);
+        assert.deepEqual(axis.annotatedTicks(), annotatedTicks, "can set the annotated ticks");
+      });
+
+      it("returns the axis when setting the annotated ticks", () => {
+        let axis = new Plottable.Axis(new Plottable.Scale<{}, number>(), "bottom");
+        assert.strictEqual(axis.annotatedTicks([]), axis, "setting the annotated ticks returns calling axis");
+      });
+    });
+
+    describe("formatting annotation ticks", () => {
+      it("uses the identity formatter by default", () => {
+        let axis = new Plottable.Axis(new Plottable.Scale<{}, number>(), "bottom");
+        let annotationFormatter = axis.annotationFormatter();
+        let identityFormatter = Plottable.Formatters.identity();
+        let test = "testString";
+        assert.strictEqual(annotationFormatter(test), identityFormatter(test), "uses the identity formatter by default");
+      });
+
+      it("can set the annotation formatter", () => {
+        let axis = new Plottable.Axis(new Plottable.Scale<{}, number>(), "bottom");
+        let annotationFormatter = (d: any) => "";
+        axis.annotationFormatter(annotationFormatter);
+        let test = "testString";
+        assert.strictEqual(annotationFormatter(test), axis.annotationFormatter()(test), "can set the annotated ticks");
+      });
+
+      it("returns the axis when setting the annotation formatter", () => {
+        let axis = new Plottable.Axis(new Plottable.Scale<{}, number>(), "bottom");
+        assert.strictEqual(axis.annotationFormatter(() => ""), axis, "setting the annotation formatter returns calling axis");
+      });
+    });
+
+    describe("rendering annotations", () => {
+      describe("label content", () => {
+        it("does not render annotations by default", () => {
```

Comment:
There are essentially 2 aspects to our library.

One is setting up the state on the javascript object, and the other is rendering to the DOM.

The previous test is checking the state on the javascript object but this one is checking that this changed state is reflected in the DOM.